Drivers/SensorFusion/Src/sensor_fusion.cpp

```diff
@@ -51,6 +51,9 @@ struct SFIterationData_t{
     float prevP[NUM_KALMAN_VALUES*NUM_KALMAN_VALUES];
 };
 
+// Globals
+static SemaphoreHandle_t SF_mutex;
```

Comment:
Any particular reason this needs to be global?

Reply:
global mutex implies that the same mutex or the lock can exist and work for different functions(read & write). that is what i thought

Reply:
My bad. I thought this was a C++ class